client/client.go

```diff
@@ -180,21 +180,33 @@ func (e AuthorizationError) Error() string {
 	return fmt.Sprintf("cannot add authorization: %v", e.error)
 }
 
-type ConnectionError struct{ error }
+type ConnectionError struct{ Err error }
 
 func (e ConnectionError) Error() string {
 	var errStr string
-	switch e.error {
+	switch e.Err {
 	case context.DeadlineExceeded:
 		errStr = "timeout exceeded while waiting for response"
 	case context.Canceled:
 		errStr = "request canceled"
 	default:
-		errStr = e.error.Error()
+		errStr = e.Err.Error()
 	}
 	return fmt.Sprintf("cannot communicate with server: %s", errStr)
 }
 
+func (e ConnectionError) Is(err error) bool {
+	return IsConnectionError(err)
+}
+
+func IsConnectionError(err error) bool {
+	if _, ok := err.(ConnectionError); ok {
+		return true
+	}
+	_, ok := err.(*ConnectionError)
+	return ok
```

Comment:
I removed this one entirely now, this can be done better with "xerrors.As()" I think (I use that in snap_cmd_known.go now to fallback). But feedback welcome.